errors.go

```diff
@@ -32,3 +32,14 @@ func (e *RequestError) Error() string {
 	}
 	return msg
 }
+
+// InvalidRequestError is returned when builder errors on creating API request object.
```

Comment:
I don't think dedicated error structs like these are so useful, since there's no reasonable way to handle them (e.g. and retry). I think just using `fmt.Errorf()` and `errors.New()` makes more sense. Error structs should be used when there is more useful information to glean from an error.